sources/early-boot-config/early-boot-config-provider/src/settings.rs

```diff
@@ -1,14 +1,14 @@
 //! The settings module owns the `SettingsJson` struct which contains the JSON settings data being
 //! sent to the API.
 
-use serde::Serialize;
+use serde::{Deserialize, Serialize};
 use snafu::{OptionExt, ResultExt};
 
 /// SettingsJson represents a change that a provider would like to make in the API.
-#[derive(Debug)]
```

Comment:
Just to help me understand, how was this used before if it did not derive `Serialize` or `Deserialize`?

Reply:
when providers existed in the same crate as `early-boot-config`, they could just return this struct, with the settings changes pre-serialized in the `json` field, which at the time was a String. we only need `Serialize` and `Deserialize` now because we're using this as the return format of a provider binary